x/twap/listeners_test.go

```diff
@@ -50,6 +50,7 @@ func (s *TestSuite) TestSwapAndEndBlockTriggeringSave() {
 	s.Commit() // clear transient store
 	// Now on a clean state after a create pool
 	s.Require().Equal(baseTime.Add(time.Second), s.Ctx.BlockTime())
+	s.RunBasicJoin(poolId)
```

Comment:
Note to reviewer: This is needed to test the JoinPool twap hook (mutation tests caught this!)

Reply:
Err, this should be its own test. We need to test on a clean instance, a join in a block will trigger the end tracking.

Reply:
Done!